src/storage/txn/actions/acquire_pessimistic_lock.rs

```diff
@@ -25,6 +25,11 @@ pub fn acquire_pessimistic_lock<S: Snapshot>(
         crate::storage::mvcc::txn::make_txn_error(err, &key, reader.start_ts).into()
     ));
 
+    // Update max_ts for Insert operation to guarante linearizability and snapshot isolation
```

Comment:
Could you also add a unit test for this?

Reply:
Done. 
PTAL